frontend/src/metabase/auth/containers/LoginApp.jsx

```diff
@@ -51,13 +54,49 @@ export default class LoginApp extends Component {
     }
 
     componentDidMount() {
+
         this.validateForm();
+
+        const { loginGoogle, onChangeLocation } = this.props;
+
+        let ssoLoginButton = findDOMNode(this.refs.ssoLoginButton);
+
+        function attachGoogleAuth() {
+            // if gapi isn't loaded yet then wait 100ms and check again. Keep doing this until we're ready
+            if (!window.gapi) {
+                window.setTimeout(attachGoogleAuth, 100);
+                return;
+            }
+            try {
+                window.gapi.load('auth2', () => {
+                  let auth2 = window.gapi.auth2.init({
+                      client_id: Settings.get('google_auth_client_id'),
+                      cookiepolicy: 'single_host_origin',
+                  });
+                  auth2.attachClickHandler(ssoLoginButton, {},
+                      (googleUser) => loginGoogle(googleUser, onChangeLocation),
+                      (err) => console.error('err', err)
```

Comment:
I'm looking at the docs to try and figure out what kind(s) of errors would cause this callback to get called. It might be worth doing a generic "there was a problem logging in" or something more specific depending on what the error is but I want to know a bit more before we settle on that route.